rasa_nlu/training_data/data_loader.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+import json
+
+from rasa_nlu import utils
+from rasa_nlu.training_data.formats import MarkdownReader, WitReader, LuisReader
+from rasa_nlu.training_data import TrainingData
+
+logger = logging.getLogger(__name__)
+
+# Different supported file formats and their identifier
+WIT_FILE_FORMAT = "wit"
+DIALOGFLOW_FILE_FORMAT = "dialogflow"
+LUIS_FILE_FORMAT = "luis"
+RASA_FILE_FORMAT = "rasa_nlu"
+UNK_FILE_FORMAT = "unk"
+MARKDOWN_FILE_FORMAT = "md"
+
+
+class DataLoader(object):
+
+    @classmethod
+    def from_wit_file(cls, filename):
+        cls._from_file(WitReader, filename)
+
+    @classmethod
+    def from_luis_file(cls, filename):
+        cls._from_file(LuisReader, filename)
+
+    @classmethod
+    def from_markdown_file(cls, filename):
+        return cls._from_file(MarkdownReader, filename)
+
+    @classmethod
+    def from_markdown_string(cls, s):
+        return cls._from_string(MarkdownReader, s)
+
+    @staticmethod
+    def _from_file(reader_cls, filename, **kwargs):
+        reader = reader_cls()
+        return reader.read(filename, **kwargs)
+
+    @staticmethod
+    def _from_string(reader_cls, s, **kwargs):
+        reader = reader_cls()
+        return reader.reads(s, **kwargs)
+
+    @staticmethod
+    def _from_json(reader_cls, js, **kwargs):
+        reader = reader_cls()
+        return reader.read_from_json(js, **kwargs)
+
+    @classmethod
+    def load(cls, resource_name, language='en', fformat=None):
```

Comment:
This should probably be attached to `TrainingData` class

Reply:
I think it could go there, yes. I separated it, because: 1) I felt that data handling finding multiple data files and guessing file formats etc. is a separate concern and 2) the TrainingData class already being quite large itself. Wondering if I should put all the DataLoader functions into the modules __init__.py. Thoughts?